### PR TITLE
Fix the Import behaviour of nvidia-ml-py ( pynvml )

### PR DESCRIPTION
**Pull request by sagewhocodes**
This PR fixes the Import behaviour of `pynvml` to only be imported in case of GPU enabled environments.

### CODE REVIEW

File: ivy/functional/ivy/device.py [hunk 1]
```diff
@@ -6,15 +6,26 @@
...
-    pynvml.nvmlInit()
-except pynvml.NVMLError:
+    import pynvml
+
+    pynvml_installed = True
+except ImportError:
+    # nvidia-ml-py (pynvml) is not installed in CPU Dockerfile.
+    pynvml_installed = False
     pass
+
```
Comment (Dsantra92) [inline ivy/functional/ivy/device.py:22]: merge into one try statement? `pynvml_installed` can be skipped.

File: ivy/functional/ivy/device.py [hunk 2]
```diff
@@ -6,14 +6,20 @@
...
-    pynvml.nvmlInit()
-except pynvml.NVMLError:
+    import pynvml
+
+    try:
+        pynvml.nvmlInit()
+    except pynvml.NVMLError:
+        pass
+except ImportError:
+    # nvidia-ml-py (pynvml) is not installed in CPU Dockerfile.
```
Comment (Dsantra92) [inline ivy/functional/ivy/device.py:22]: Could you add a warning or throw an import error when GPU is present but pynvml is not installed? I feel it's better to show import warnings during initialization, than during actual work.

**Review by Dsantra92 — CHANGES_REQUESTED** [review]
I have one thing to add here:
We need to check if intelligent tests are run on GPU. If not, we can safely remove pyvnml installation. I think @RashulChutani can help us here.

**Comment by RashulChutani** [comment]
> I have one thing to add here:
> We need to check if intelligent tests are run on GPU. If not, we can safely remove pyvnml installation. I think @RashulChutani can help us here.

They are not currently run on GPU. We will be extending to GPUs soon but we will pass the GPU using the --device argument.

**Review by Dsantra92 — APPROVED** [review]
LGTM

**Review by VedPatwardhan — APPROVED** [review]
lgtm!

**Review by RashulChutani — APPROVED** [review]
Looks good to me!

**Review by CatB1t — COMMENTED** [review]
Just looking at the git  blame, the import seems to be there for a little while, also `pynvml` is listed in the requirements but it fails these days, maybe worth looking why it started failing recently? some tests internally use `pynvml` so the try except block won't be really helpful in making sure these doesn't run.

**Comment by sagewhocodes** [comment]
> Just looking at the git blame, the import seems to be there for a little while, also `pynvml` is listed in the requirements but it fails these days, maybe worth looking why it started failing recently?

We actually made changes to the CPU Dockerfile such that it excludes nvidia based packages while building the CPU only version, however `pynvml` was always imported regardless if there was a GPU found or not. 
I have run tests under `ivy_tests` folder and didn't find any related errors, do I need to run tests from some other directory too?

**Review by CatB1t — APPROVED** [review]
lgtm!

**Pull request merged**